sei-db/sc/memiavl/multitree.go

```diff
@@ -401,28 +405,74 @@ func (t *MultiTree) Catchup(stream types.Stream[proto.ChangelogEntry], endVersio
 }
 
 func (t *MultiTree) WriteSnapshot(ctx context.Context, dir string, wp *pond.WorkerPool) error {
+	t.logger.Info("starting snapshot write", "trees", len(t.trees))
+
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil { //nolint:gosec
 		return err
 	}
 
-	// write the snapshots in parallel and wait all jobs done
-	group, _ := wp.GroupContext(ctx)
+	// Write EVM first to avoid disk I/O contention, then parallel
+	return t.writeSnapshotPriorityEVM(ctx, dir, wp)
+}
+
+// writeSnapshotPriorityEVM writes EVM tree first, then others in parallel
+// Best strategy: reduces disk I/O contention for the largest tree
+func (t *MultiTree) writeSnapshotPriorityEVM(ctx context.Context, dir string, wp *pond.WorkerPool) error {
```

Comment:
Beyond the scope of this PR: i recommend removing dependency to `pond` custom library in favour of what `sync` package offers. ErrorGroup for example would fit here. concurrency utilities in Go have come a long way.